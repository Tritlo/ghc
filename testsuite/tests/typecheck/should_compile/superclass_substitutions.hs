module SuperclassSubs where

mymax :: Ord a => [a] -> a
mymax = _
