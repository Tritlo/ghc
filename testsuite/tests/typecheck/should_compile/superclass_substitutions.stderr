
superclass_substitutions.hs:4:9: warning: [-Wtyped-holes (in -Wdefault)]
    • Found hole: _ :: [a] -> a
      Where: ‘a’ is a rigid type variable bound by
               the type signature for:
                 mymax :: forall a. Ord a => [a] -> a
               at superclass_substitutions.hs:3:1-26
    • In the expression: _
      In an equation for ‘mymax’: mymax = _
    • Relevant bindings include
        mymax :: [a] -> a (bound at superclass_substitutions.hs:4:1)
      Valid substitutions include
        mymax :: [a] -> a (bound at superclass_substitutions.hs:4:1)
        maximum :: forall (t :: * -> *).
                   Foldable t =>
                   forall a. Ord a => t a -> a
          (imported from ‘Prelude’ at superclass_substitutions.hs:1:8-21
           (and originally defined in ‘Data.Foldable’))
        minimum :: forall (t :: * -> *).
                   Foldable t =>
                   forall a. Ord a => t a -> a
          (imported from ‘Prelude’ at superclass_substitutions.hs:1:8-21
           (and originally defined in ‘Data.Foldable’))
        head :: forall a. [a] -> a
          (imported from ‘Prelude’ at superclass_substitutions.hs:1:8-21
           (and originally defined in ‘GHC.List’))
        last :: forall a. [a] -> a
          (imported from ‘Prelude’ at superclass_substitutions.hs:1:8-21
           (and originally defined in ‘GHC.List’))
